### ISSUE TITLE
Peer dependencies problem with ts-node with yarn v2

### ISSUE THREAD

**Issue by myndelx** [intuit/auto#1890]
**Describe the bug**

When installing the last version (10.20.4) of auto, some problems with peer dependencies appear:

```
$ yarn install
➤ YN0000: ┌ Resolution step
➤ YN0002: │ @auto-it/core@npm:10.20.4 doesn't provide typescript (pdd9be), requested by ts-node
➤ YN0002: │ @endemolshinegroup/cosmiconfig-typescript-loader@npm:3.0.2 [f111c] doesn't provide typescript (p2b965), requested by ts-node
➤ YN0000: │ Some peer dependencies are incorrectly met; run yarn explain peer-requirements <hash> for details, where <hash> is the six-letter p-prefixed code
➤ YN0000: └ Completed in 0s 285ms
```

**To Reproduce**

With yarn v2:

```
$ yarn add -D auto
```

**Expected behavior**

We shouldn't have warnings about peer dependencies requirements not met.

**Environment information:**

```txt
Environment Information:

"auto" version: v10.20.4
"git"  version: v2.25.1
"node" version: v10.21.0
GHE version:    v2.22.3
```

**Additional context**

This issue appears since the v10.20.0 release with yarn v2.

From what I see, the `ts-node` package has a peer dependency of `typescript` and `ts-node` is a direct dependency of the `@auto-it/core`package but it does not provide `typescript`, so the peer dependency couldn't be met.

As a solution, maybe the `@auto-it/core` package should use the `peerDependenciesMeta` field?

```json
"peerDependenciesMeta": {
  "typescript": {
    "optional": true
  }
}
```

We also got the exact same issue with the `@endemolshinegroup/cosmiconfig-typescript-loader` package but an issue has already been created: https://github.com/EndemolShineGroup/cosmiconfig-typescript-loader/issues/147

**Comment by hipstersmoothie** [comment]
Is that all I have to do? If you submit a PR to fix the issue I'll merge it ASAP

**Comment by myndelx** [comment]
Yes, it should be enough to fix the warning as `typescript` is required by `ts-node`, but it's optional for using `auto`.

**Comment by adierkens** [comment]
:rocket: Issue was released in [`v10.20.5`](https://github.com/intuit/auto/releases/tag/v10.20.5) :rocket:
